python/cuml/dask/common/comms.py

```diff
@@ -178,6 +175,26 @@ def _func_init_nccl(sessionId, uniqueId):
         print("An error occurred initializing NCCL!")
 
 
+class ListenerThread(threading.Thread):
+    def __init__(self, verbose):
+        threading.Thread.__init__(self)
+        self.listener = ucp.create_listener(_connection_func)
+        self.port = self.listener.port
+        self.verbose = verbose
+
+    def run(self):
+        if self.verbose:
+            print("Running listener thread")
+        while not self.listener.closed():
```

Comment:
Is this going to become a lingering thread that keeps the process alive if you control-c it or otherwise kill it since there will be nobody alive to call the `close` method? could make it a setDaemon thread if so